crates/router/src/core/refunds.rs

```diff
@@ -295,9 +295,11 @@ pub async fn sync_refund_with_gateway(
 
     logger::debug!(refund_retrieve_router_data=?router_data);
 
-    match add_access_token_result.access_token_result {
-        Ok(access_token) => router_data.access_token = access_token,
-        Err(connector_error) => router_data.response = Err(connector_error),
+    if add_access_token_result.connector_supports_access_token {
+        match add_access_token_result.access_token_result {
+            Ok(access_token) => router_data.access_token = access_token,
+            Err(connector_error) => router_data.response = Err(connector_error),
+        }
```

Comment:
Is this extra check necessary, now that you have made the changes to not read from response in routerdata.

Reply:
Currently, @jagan-jaya is making a change for `payu`, `cybersource` and `globalpay` which still uses the response for fetching the connector_refund_id, we discussed that once this PR get's merged we can raise another PR to remove those changes too. Till then, to keep the refund_retrieve_core from breaking. This change needs to be kept.

CC: @jagan-jaya